lib/collection/src/collection.rs

```diff
@@ -52,10 +73,10 @@ impl Collection {
     ) -> CollectionResult<UpdateResult> {
         let operation_id = self.wal.lock().write(&operation)?;
 
-        let upd = self.updater.clone();
         let sndr = self.update_sender.clone();
+        let segments = self.segments.clone();
         let update_future = async move {
-            let res = upd.update(operation_id, operation);
+            let res = SimpleSegmentUpdater::update(segments.deref(), operation_id, operation);
```

Comment:
My original intention was to make search strategy replaceable. So that searcher implementation could be replaced by other implementation of the `SegmentSearcher` trait. That is why it is called `Simple...`.  Now it seems like `SegmentSearcher` and `SegmentUpdater` are redundant and functions could be extended in different ways.

Removing `searcher` from collection is basically a step in this simplification. So I propose to also remove `SegmentSearcher` and `SegmentUpdater` here as well.

Reply:
We could move `searcher` and `updater` to `TableOfContent` level where we can use them as trait objects.  I was trying to do it in this PR, but it increased the number of changes and made the PR too big. In that case, we don't give up the idea of replaceable searcher and updater and also the collection completely separates from them. WDYT?